src/Microsoft.TemplateEngine.Abstractions/GlobalSettings/IGlobalSettings.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.TemplateEngine.Abstractions.GlobalSettings
+{
+    /// <summary>
+    /// Interface that represents loading/storing data into settings.json file.
+    /// That is shared between multiple different hosts of TemplateEngine.
+    /// </summary>
+    public interface IGlobalSettings
+    {
+        /// <summary>
+        /// Triggered every time when settings change.
```

Comment:
For now yes, it is only list of globally installed packages. But the plan is to extend it to also have some global settings, for example "Default Language" set to C# (now it's hardcoded), whether to check for template packages update when instantiating templates (host wide) etc,

Reply:
Thank you, I'm trying to imagine how the event is used in the future when there's more settings. Do you think it would make sense if the callback gave an indication of what changed? Now it's just an `Action` so the subscriber doesn't know why it was triggered and to figure that out has to keep a previous version of the setting of interest to compare it to.

Reply:
If "the plan is to extend it", then making this an abstract class would be better. We can add new members without breaking anybody. As far as I know, that is a common approach in runtime repo as well.

Reply:
After discussion with @DavidKarlas we decided to make this interface internal, it is moved to `Edge` and won't be accessible anymore. Likely it would be only limited to packages and for settings like above we implement other approach.